AnaProd/tasks.py

```diff
@@ -116,4 +161,80 @@ def getBranches(samples, central_nanoAOD_path, max_files_per_job):
+        return [ AnaTupleTask.req(self, branch=sample_id, max_runtime=AnaTupleTask.max_runtime._default) ]
+
+    def create_branch_map(self):
+        self.load_sample_configs()
+        return DataMergingTask.getBranches(self.samples, self.central_anaTuples_path(), self.max_files_per_job)
+
+    def output(self, force_pre_output=False):
+        sample_id, sample_name, sample_type, split_idx, input_files = self.branch_data
+        outFileName = 'nano_merged.root'
+        out = os.path.join(self.central_anaTuples_path(), sample_name,outFileName)
+        return law.LocalFileTarget(out)
+
+    def run(self):
+        job_home, remove_job_home = self.law_job_home()
+        sample_id, sample_name, sample_type, split_idx, input_files = self.branch_data
+        producer_mergentuples = os.path.join(self.ana_path(), 'AnaProd', 'MergeNtuples.py')
+        anaTuple_dir = os.path.join(self.central_anaTuples_path(), sample_name)
+        finalFile = self.output().path
+        if(os.path.exists(finalFile)):
+            os.remove(finalFile)
+        sh_call([ 'python3', producer_mergentuples,  '--inputDir', anaTuple_dir, '--outputFile', finalFile], env=self.cmssw_env(),verbose=1)
+        if os.path.exists(finalFile):
+            for anaFile in os.listdir(anaTuple_dir):
+                anaFile_complete = os.path.join(anaTuple_dir, anaFile)
+                print(anaFile_complete)
+                #if anaFile_complete == finalFile : continue
+                #os.remove(anaFile_complete)
+        print(f'anaTuple for sample {sample_name}  split_idx={split_idx} is created in {anaTuple_dir}')
+
+    @staticmethod
+    def getInputFiles(central_anatuple_path, sample_name):
+        inDir = os.path.join(central_anatuple_path, sample_name)
+        input_files = []
+        for root, dirs, files in os.walk(inDir):
+            for file in files:
+                if file.endswith('.root') and not file.startswith('.'):
+                    input_files.append(os.path.join(root, file))
+        return list(sorted(input_files))
+
+
+    def getBranches(samples, central_anaTuples_path, max_files_per_job):
+        n = 0
+        branches = {}
+        for sample_id, sample_name in enumerate(sorted(samples.keys())):
+            input_files = DataMergingTask.getInputFiles(central_anaTuples_path, sample_name)
+            if len(input_files) == 0:
+                raise RuntimeError(f"DataMergingTask: no input files found for {sample_name}")
+            split_idx = 0
+            while True:
+                start_idx, stop_idx = split_idx * max_files_per_job, (split_idx + 1) * max_files_per_job
+                branches[n] = (sample_id, sample_name, samples[sample_name]['sampleType'], split_idx,
+                               input_files[start_idx:stop_idx])
+                split_idx += 1
+                n += 1
+                if stop_idx >= len(input_files):
+                    break
         return branches
```

Comment:
reimplement
- there is only one branch for merging data
- don't rely on input files to define branch_data, just list of sample names